# Projeto A3 Análise de Dados (Construção do conjunto de dados)

## Identificação da Equipe


*   **IAN MENEZES SILVA PEREIRA CARVALHO**.

*   **AMANDA NASCIMENTO CAMPOS**.

*   **BRENO BARRETO CAMPOS**.

*   **GABRIEL FAGUNDES DE ALMEIDA**.

## Montando o conjunto de dados

### Imports

In [ ]:
import pandas as pd
import numpy as np

### Carregando e checando todos os conjuntos de dados

Neste documento, iremos mesclar **quatro** datasets em um único conjunto de dados.

Os datasets escolhidos são sobre amostras de filmes e séries das plataformas: **"Amazon Prime"**, **"Netflix"**, **"Disney+"** e "**Hulu"**.

Abaixo, carregaremos e visualizaremos todos os datasets a serem mesclados.

Carregando os dados dos filmes e séries da Amazon Prime:

In [ ]:
dados_amazon = pd.read_csv("amazon_prime_titles.csv")

Visualizando os dados dos filmes e séries da Amazon Prime:

In [ ]:
dados_amazon.head()

show_id   type                 title        director  \
0      s1  Movie   The Grand Seduction    Don McKellar   
1      s2  Movie  Take Care Good Night    Girish Joshi   
2      s3  Movie  Secrets of Deception     Josh Webber   
3      s4  Movie    Pink: Staying True  Sonia Anderson   
4      s5  Movie         Monster Maker    Giles Foster   

                                                cast         country  \
0     Brendan Gleeson, Taylor Kitsch, Gordon Pinsent          Canada   
1   Mahesh Manjrekar, Abhay Mahajan, Sachin Khedekar           India   
2  Tom Sizemore, Lorenzo Lamas, Robert LaSardo, R...   United States   
3  Interviews with: Pink, Adele, Beyoncé, Britney...   United States   
4  Harry Dean Stanton, Kieran O'Brien, George Cos...  United Kingdom   

       date_added  release_year rating duration                listed_in  \
0  March 30, 2021          2014    NaN  113 min            Comedy, Drama   
1  March 30, 2021          2018    13+  110 min     Drama, International   
2  March 30, 2021          2017    NaN   74 min  Action, Drama, Suspense   
3  March 30, 2021          2014    NaN   69 min              Documentary   
4  March 30, 2021          1989    NaN   45 min           Drama, Fantasy   

                                         description  
0  A small fishing village must procure a local d...  
1  A Metro Family decides to fight a Cyber Crimin...  
2  After a man discovers his wife is cheating on ...  
3  Pink breaks the mold once again, bringing her ...  
4  Teenage Matt Banting wants to work with a famo...

Carregando os dados dos filmes e séries da Disney+:

In [ ]:
dados_disney_plus = pd.read_csv("disney_plus_titles.csv")

Visualizando os dados dos filmes e séries da Disney+:

In [ ]:
dados_disney_plus.head()

show_id     type                                             title  \
0      s1    Movie  Duck the Halls: A Mickey Mouse Christmas Special   
1      s2    Movie                            Ernest Saves Christmas   
2      s3    Movie                      Ice Age: A Mammoth Christmas   
3      s4    Movie                        The Queen Family Singalong   
4      s5  TV Show                             The Beatles: Get Back   

                            director  \
0  Alonso Ramirez Ramos, Dave Wasson   
1                        John Cherry   
2                       Karen Disher   
3                    Hamish Hamilton   
4                                NaN   

                                                cast        country  \
0  Chris Diamantopoulos, Tony Anselmo, Tress MacN...            NaN   
1           Jim Varney, Noelle Parker, Douglas Seale            NaN   
2  Raymond Albert Romano, John Leguizamo, Denis L...  United States   
3  Darren Criss, Adam Lambert, Derek Hough, Alexa...            NaN   
4  John Lennon, Paul McCartney, George Harrison, ...            NaN   

          date_added  release_year rating  duration  \
0  November 26, 2021          2016   TV-G    23 min   
1  November 26, 2021          1988     PG    91 min   
2  November 26, 2021          2011   TV-G    23 min   
3  November 26, 2021          2021  TV-PG    41 min   
4  November 25, 2021          2021    NaN  1 Season   

                       listed_in  \
0              Animation, Family   
1                         Comedy   
2      Animation, Comedy, Family   
3                        Musical   
4  Docuseries, Historical, Music   

                                         description  
0   Join Mickey and the gang as they duck the halls!  
1  Santa Claus passes his magic bag to a new St. ...  
2          Sid the Sloth is on Santa's naughty list.  
3               This is real life, not just fantasy!  
4  A three-part documentary from Peter Jackson ca...

Carregando os dados dos filmes e séries da Netflix:

In [ ]:
dados_netflix = pd.read_csv("netflix_titles.csv")

Visualizando os dados dos filmes e séries da Netflix:

In [ ]:
dados_netflix.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

Carregando os dados dos filmes e séries da Hulu:

In [ ]:
dados_hulu = pd.read_csv("hulu_titles.csv")

Visualizando os dados dos filmes e séries da Hulu:

In [ ]:
dados_hulu.head()

show_id   type                           title director  cast country  \
0      s1  Movie  Ricky Velez: Here's Everything      NaN   NaN     NaN   
1      s2  Movie                    Silent Night      NaN   NaN     NaN   
2      s3  Movie                    The Marksman      NaN   NaN     NaN   
3      s4  Movie                            Gaia      NaN   NaN     NaN   
4      s5  Movie                        Settlers      NaN   NaN     NaN   

         date_added  release_year rating duration                  listed_in  \
0  October 24, 2021          2021  TV-MA      NaN           Comedy, Stand Up   
1  October 23, 2021          2020    NaN   94 min     Crime, Drama, Thriller   
2  October 23, 2021          2021  PG-13  108 min           Action, Thriller   
3  October 22, 2021          2021      R   97 min                     Horror   
4  October 22, 2021          2021    NaN  104 min  Science Fiction, Thriller   

                                         description  
0  ​Comedian Ricky Velez bares it all with his ho...  
1  Mark, a low end South London hitman recently r...  
2  A hardened Arizona rancher tries to protect an...  
3  A forest ranger and two survivalists with a cu...  
4  Mankind's earliest settlers on the Martian fro...

### Adicionando uma coluna identificadora em cada conjunto

Para conseguirmos identificar de qual conjunto de dados uma determinada instância veio, iremos adicionar um atributo novo em cada dataset para informar a origem das instâncias.

Resolveremos esse problema utilizando a função "np.full" da biblioteca "numpy".

Ela é capaz de gerar um vetor preenchido com um determinado valor.

Criaremos 4 vetores e os preencheremos com os valores: "Amazon Prime", "Netflix", "Disney+" e "Hulu".

Esses vetores se chamarão: "identificador_amazon", "identificador_disney_plus", "identificador_netflix" e "identificador_hulu".

Cada um dos vetores terá o tamanho de seu respectivo dataset, como observado nos argumentos abaixo.

In [ ]:
identificador_amazon = np.full(len(dados_amazon), "Amazon Prime")

In [ ]:
identificador_disney_plus = np.full(len(dados_disney_plus), "Disney+")

In [ ]:
identificador_netflix = np.full(len(dados_netflix), "Netflix")

In [ ]:
identificador_hulu = np.full(len(dados_hulu), "Hulu")

O próximo passo é: Iremos transformar esses vetores em colunas, unindo cada um deles a seu respectivo DataFrame.

Iremos nomear a coluna que guardará os vetores identificadores de "dataset_de_origem".

In [ ]:
dados_amazon["dataset_de_origem"] = identificador_amazon

In [ ]:
dados_disney_plus["dataset_de_origem"] = identificador_disney_plus

In [ ]:
dados_netflix["dataset_de_origem"] = identificador_netflix

In [ ]:
dados_hulu["dataset_de_origem"] = identificador_hulu

Por último, vamos verificar se todos os datasets realmente acolheram uma coluna que identifica suas amostras.

Primerio, o da Amazon Prime:

In [ ]:
display(dados_amazon)

show_id     type                    title         director  \
0         s1    Movie      The Grand Seduction     Don McKellar   
1         s2    Movie     Take Care Good Night     Girish Joshi   
2         s3    Movie     Secrets of Deception      Josh Webber   
3         s4    Movie       Pink: Staying True   Sonia Anderson   
4         s5    Movie            Monster Maker     Giles Foster   
...      ...      ...                      ...              ...   
9663   s9664    Movie      Pride Of The Bowery  Joseph H. Lewis   
9664   s9665  TV Show            Planet Patrol              NaN   
9665   s9666    Movie                  Outpost     Steve Barker   
9666   s9667  TV Show  Maradona: Blessed Dream              NaN   
9667   s9668    Movie              Harry Brown    Daniel Barber   

                                                   cast         country  \
0        Brendan Gleeson, Taylor Kitsch, Gordon Pinsent          Canada   
1      Mahesh Manjrekar, Abhay Mahajan, Sachin Khedekar           India   
2     Tom Sizemore, Lorenzo Lamas, Robert LaSardo, R...   United States   
3     Interviews with: Pink, Adele, Beyoncé, Britney...   United States   
4     Harry Dean Stanton, Kieran O'Brien, George Cos...  United Kingdom   
...                                                 ...             ...   
9663                           Leo Gorcey, Bobby Jordan             NaN   
9664  DICK VOSBURGH, RONNIE STEVENS, LIBBY MORRIS, M...             NaN   
9665  Ray Stevenson, Julian Wadham, Richard Brake, M...             NaN   
9666  Esteban Recagno, Ezequiel Stremiz, Luciano Vit...             NaN   
9667  Michael Caine, Emily Mortimer, Joseph Gilgun, ...             NaN   

          date_added  release_year rating   duration                listed_in  \
0     March 30, 2021          2014    NaN    113 min            Comedy, Drama   
1     March 30, 2021          2018    13+    110 min     Drama, International   
2     March 30, 2021          2017    NaN     74 min  Action, Drama, Suspense   
3     March 30, 2021          2014    NaN     69 min              Documentary   
4     March 30, 2021          1989    NaN     45 min           Drama, Fantasy   
...              ...           ...    ...        ...                      ...   
9663             NaN          1940     7+     60 min                   Comedy   
9664             NaN          2018    13+  4 Seasons                 TV Shows   
9665             NaN          2008      R     90 min                   Action   
9666             NaN          2021  TV-MA   1 Season            Drama, Sports   
9667             NaN          2010      R    103 min  Action, Drama, Suspense   

                                            description dataset_de_origem  
0     A small fishing village must procure a local d...      Amazon Prime  
1     A Metro Family decides to fight a Cyber Crimin...      Amazon Prime  
2     After a man discovers his wife is cheating on ...      Amazon Prime  
3     Pink breaks the mold once again, bringing her ...      Amazon Prime  
4     Teenage Matt Banting wants to work with a famo...      Amazon Prime  
...                                                 ...               ...  
9663  New York City street principles get an East Si...      Amazon Prime  
9664  This is Earth, 2100AD - and these are the adve...      Amazon Prime  
9665  In war-torn Eastern Europe, a world-weary grou...      Amazon Prime  
9666  The series tells the story of Diego Maradona, ...      Amazon Prime  
9667  Harry Brown, starring two-time Academy Award w...      Amazon Prime  

[9668 rows x 13 columns]

Depois, o da Disney+:

In [ ]:
display(dados_disney_plus)

show_id     type                                             title  \
0         s1    Movie  Duck the Halls: A Mickey Mouse Christmas Special   
1         s2    Movie                            Ernest Saves Christmas   
2         s3    Movie                      Ice Age: A Mammoth Christmas   
3         s4    Movie                        The Queen Family Singalong   
4         s5  TV Show                             The Beatles: Get Back   
...      ...      ...                                               ...   
1445   s1446    Movie                          X-Men Origins: Wolverine   
1446   s1447    Movie    Night at the Museum: Battle of the Smithsonian   
1447   s1448    Movie                                   Eddie the Eagle   
1448   s1449    Movie                              Bend It Like Beckham   
1449   s1450    Movie             Captain Sparky vs. The Flying Saucers   

                               director  \
0     Alonso Ramirez Ramos, Dave Wasson   
1                           John Cherry   
2                          Karen Disher   
3                       Hamish Hamilton   
4                                   NaN   
...                                 ...   
1445                         Gavin Hood   
1446                         Shawn Levy   
1447                    Dexter Fletcher   
1448                    Gurinder Chadha   
1449                        Mark Waring   

                                                   cast  \
0     Chris Diamantopoulos, Tony Anselmo, Tress MacN...   
1              Jim Varney, Noelle Parker, Douglas Seale   
2     Raymond Albert Romano, John Leguizamo, Denis L...   
3     Darren Criss, Adam Lambert, Derek Hough, Alexa...   
4     John Lennon, Paul McCartney, George Harrison, ...   
...                                                 ...   
1445  Hugh Jackman, Liev Schreiber, Danny Huston, wi...   
1446  Ben Stiller, Amy Adams, Owen Wilson, Hank Azar...   
1447  Tom Costello, Jo Hartley, Keith Allen, Dickon ...   
1448  Parminder Nagra, Keira Knightley, Jonathan Rhy...   
1449                                      Charlie Tahan   

                                     country          date_added  \
0                                        NaN   November 26, 2021   
1                                        NaN   November 26, 2021   
2                              United States   November 26, 2021   
3                                        NaN   November 26, 2021   
4                                        NaN   November 25, 2021   
...                                      ...                 ...   
1445           United States, United Kingdom        June 4, 2021   
1446                   United States, Canada       April 2, 2021   
1447  United Kingdom, Germany, United States   December 18, 2020   
1448  United Kingdom, Germany, United States  September 18, 2020   
1449                           United States       April 1, 2020   

      release_year rating  duration  \
0             2016   TV-G    23 min   
1             1988     PG    91 min   
2             2011   TV-G    23 min   
3             2021  TV-PG    41 min   
4             2021    NaN  1 Season   
...            ...    ...       ...   
1445          2009  PG-13   108 min   
1446          2009     PG   106 min   
1447          2016  PG-13   107 min   
1448          2003  PG-13   112 min   
1449          2012   TV-G     2 min   

                                          listed_in  \
0                                 Animation, Family   
1                                            Comedy   
2                         Animation, Comedy, Family   
3                                           Musical   
4                     Docuseries, Historical, Music   
...                                             ...   
1445      Action-Adventure, Family, Science Fiction   
1446               Action-Adventure, Comedy, Family   
1447                    Biographical, Comedy, Drama   
1448                   Buddy, Comedy, Comi

Agora, o da Netflix:

In [ ]:
display(dados_netflix)

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description dataset_de_origem  
0     As her father nears the end of his life, filmm...           Netflix  
1     After crossing paths at a party, a Cape Town t...           Netflix  
2     To protect his family from a powerful drug lor...           Netflix  
3     Feuds, flirtations and toilet talk go down amo...           Netflix  
4     In a city of coaching centers known to train I...           Netflix  
...                                                 ...               ...  
8802  A political cartoonist, a crime reporter and a...           Netflix  
8803  While living alone in a spooky town, a young g...           Netflix  
8804  Looking to survive in a world taken over by zo...           Netflix  
8805  Dragged from civilian life, a former superhero...           Netflix  
8806  A scrappy but poor boy worms his way into a ty...           Net

Por último, o da Hulu:

In [ ]:
display(dados_hulu)

show_id     type                           title director  cast  \
0         s1    Movie  Ricky Velez: Here's Everything      NaN   NaN   
1         s2    Movie                    Silent Night      NaN   NaN   
2         s3    Movie                    The Marksman      NaN   NaN   
3         s4    Movie                            Gaia      NaN   NaN   
4         s5    Movie                        Settlers      NaN   NaN   
...      ...      ...                             ...      ...   ...   
3068   s3069  TV Show  Star Trek: The Original Series      NaN   NaN   
3069   s3070  TV Show              Star Trek: Voyager      NaN   NaN   
3070   s3071  TV Show                       The Fades      NaN   NaN   
3071   s3072  TV Show               The Twilight Zone      NaN   NaN   
3072   s3073  TV Show             Tokyo Magnitude 8.0      NaN   NaN   

             country        date_added  release_year rating   duration  \
0                NaN  October 24, 2021          2021  TV-MA        NaN   
1                NaN  October 23, 2021          2020    NaN     94 min   
2                NaN  October 23, 2021          2021  PG-13    108 min   
3                NaN  October 22, 2021          2021      R     97 min   
4                NaN  October 22, 2021          2021    NaN    104 min   
...              ...               ...           ...    ...        ...   
3068   United States               NaN          1966  TV-PG  3 Seasons   
3069   United States               NaN          1997  TV-PG  7 Seasons   
3070  United Kingdom               NaN          2011  TV-14   1 Season   
3071   United States               NaN          1959  TV-PG  5 Seasons   
3072           Japan               NaN          2009  TV-14   1 Season   

                                   listed_in  \
0                           Comedy, Stand Up   
1                     Crime, Drama, Thriller   
2                           Action, Thriller   
3                                     Horror   
4                  Science Fiction, Thriller   
...                                      ...   
3068             Action, Adventure, Classics   
3069      Action, Adventure, Science Fiction   
3070  Horror, International, Science Fiction   
3071     Classics, Science Fiction, Thriller   
3072             Anime, Drama, International   

                                            description dataset_de_origem  
0     ​Comedian Ricky Velez bares it all with his ho...              Hulu  
1     Mark, a low end South London hitman recently r...              Hulu  
2     A hardened Arizona rancher tries to protect an...              Hulu  
3     A forest ranger and two survivalists with a cu...              Hulu  
4     Mankind's earliest settlers on the Martian fro...              Hulu  
...                                                 ...               ...  
3068  The 23rd century adventures of Captain James T...              Hulu  
3069  Catapulted into the distant sector of the gala...              Hulu  
3070  Seventeen-year-old Paul is haunted by apocalyp...              Hulu  
3071  Rod Serling's seminal anthology series focused...              Hulu  
3072  The devastation is unleashed in the span of se...              Hulu  

[3073 rows x 13 columns]

Ao que tudo indica, a operação foi um sucesso.

### Mesclando os conjuntos de dados em um único DataFrame:

Para unir os conjuntos de dados, iremos utilizar o método "concat" da biblioteca "pandas". O argumento "axis=0" indica que queremos a união feita concatenando um conjunto embaixo do outro.

In [ ]:
dados_mesclados = pd.concat([dados_amazon, dados_disney_plus, dados_netflix, dados_hulu], axis=0)

Para conferir se a operação ocorreu com sucesso, iremos somar a quantidade de linhas de todos os conjuntos e depois comparar com a quantidade de linhas do DataFrame dos conjuntos mesclados.

In [ ]:
len(dados_amazon) + len(dados_disney_plus) + len(dados_netflix) + len(dados_hulu)

22998

In [ ]:
len(dados_mesclados)

22998

Como visto acima, a quantidade de linhas é a mesma para ambos.

Usaremos, também, a função "display" para visualizar o conjunto mesclado.

In [ ]:
display(dados_mesclados)

show_id     type                           title        director  \
0         s1    Movie             The Grand Seduction    Don McKellar   
1         s2    Movie            Take Care Good Night    Girish Joshi   
2         s3    Movie            Secrets of Deception     Josh Webber   
3         s4    Movie              Pink: Staying True  Sonia Anderson   
4         s5    Movie                   Monster Maker    Giles Foster   
...      ...      ...                             ...             ...   
3068   s3069  TV Show  Star Trek: The Original Series             NaN   
3069   s3070  TV Show              Star Trek: Voyager             NaN   
3070   s3071  TV Show                       The Fades             NaN   
3071   s3072  TV Show               The Twilight Zone             NaN   
3072   s3073  TV Show             Tokyo Magnitude 8.0             NaN   

                                                   cast         country  \
0        Brendan Gleeson, Taylor Kitsch, Gordon Pinsent          Canada   
1      Mahesh Manjrekar, Abhay Mahajan, Sachin Khedekar           India   
2     Tom Sizemore, Lorenzo Lamas, Robert LaSardo, R...   United States   
3     Interviews with: Pink, Adele, Beyoncé, Britney...   United States   
4     Harry Dean Stanton, Kieran O'Brien, George Cos...  United Kingdom   
...                                                 ...             ...   
3068                                                NaN   United States   
3069                                                NaN   United States   
3070                                                NaN  United Kingdom   
3071                                                NaN   United States   
3072                                                NaN           Japan   

          date_added  release_year rating   duration  \
0     March 30, 2021          2014    NaN    113 min   
1     March 30, 2021          2018    13+    110 min   
2     March 30, 2021          2017    NaN     74 min   
3     March 30, 2021          2014    NaN     69 min   
4     March 30, 2021          1989    NaN     45 min   
...              ...           ...    ...        ...   
3068             NaN          1966  TV-PG  3 Seasons   
3069             NaN          1997  TV-PG  7 Seasons   
3070             NaN          2011  TV-14   1 Season   
3071             NaN          1959  TV-PG  5 Seasons   
3072             NaN          2009  TV-14   1 Season   

                                   listed_in  \
0                              Comedy, Drama   
1                       Drama, International   
2                    Action, Drama, Suspense   
3                                Documentary   
4                             Drama, Fantasy   
...                                      ...   
3068             Action, Adventure, Classics   
3069      Action, Adventure, Science Fiction   
3070  Horror, International, Science Fiction   
3071     Classics, Science Fiction, Thriller   
3072             Anime, Drama, International   

                                            description dataset_de_origem  
0     A small fishing village must procure a local d...      Amazon Prime  
1     A Metro Family decides to fight a Cyber Crimin...      Amazon Prime  
2     After a man discovers his wife is cheating on ...      Amazon Prime  
3     Pink breaks the mold once again, bringing her ...      Amazon Prime  
4     Teenage Matt Banting wants to work with a famo...      Amazon Prime  
...                                                 ...               ...  
3068  The 23rd century adventures of Captain James T...              Hulu  
3069  Catapulted into the distant sector of the gala...              Hulu  
3070  Seventeen-year-old Paul is haunted by apocalyp...              Hulu  
3071  Rod Serling's seminal anthology series focused...              Hulu  
3072  The devastation is unleashed in the span of se...              Hulu  

[22998 rows x 13 columns]

O procedimento de junção dos conjuntos de dados foi bem sucedido.

## Limpando e organizando o conjunto de dados mesclado

### Eliminando as colunas "director" e "cast"

Essas duas colunas não serão utilizadas durante nossa análise e, por conta disso, para evitar que a etapa de "eliminação de valores nulos" retire amostras sem necessidade, iremos eliminá-las do conjunto mesclado.

Faremos isso usando o método "drop", da biblioteca "pandas".

In [ ]:
dados_mesclados = dados_mesclados.drop("director", axis=1)

In [ ]:
dados_mesclados = dados_mesclados.drop("cast", axis=1)

In [ ]:
display(dados_mesclados)

show_id     type                           title         country  \
0         s1    Movie             The Grand Seduction          Canada   
1         s2    Movie            Take Care Good Night           India   
2         s3    Movie            Secrets of Deception   United States   
3         s4    Movie              Pink: Staying True   United States   
4         s5    Movie                   Monster Maker  United Kingdom   
...      ...      ...                             ...             ...   
3068   s3069  TV Show  Star Trek: The Original Series   United States   
3069   s3070  TV Show              Star Trek: Voyager   United States   
3070   s3071  TV Show                       The Fades  United Kingdom   
3071   s3072  TV Show               The Twilight Zone   United States   
3072   s3073  TV Show             Tokyo Magnitude 8.0           Japan   

          date_added  release_year rating   duration  \
0     March 30, 2021          2014    NaN    113 min   
1     March 30, 2021          2018    13+    110 min   
2     March 30, 2021          2017    NaN     74 min   
3     March 30, 2021          2014    NaN     69 min   
4     March 30, 2021          1989    NaN     45 min   
...              ...           ...    ...        ...   
3068             NaN          1966  TV-PG  3 Seasons   
3069             NaN          1997  TV-PG  7 Seasons   
3070             NaN          2011  TV-14   1 Season   
3071             NaN          1959  TV-PG  5 Seasons   
3072             NaN          2009  TV-14   1 Season   

                                   listed_in  \
0                              Comedy, Drama   
1                       Drama, International   
2                    Action, Drama, Suspense   
3                                Documentary   
4                             Drama, Fantasy   
...                                      ...   
3068             Action, Adventure, Classics   
3069      Action, Adventure, Science Fiction   
3070  Horror, International, Science Fiction   
3071     Classics, Science Fiction, Thriller   
3072             Anime, Drama, International   

                                            description dataset_de_origem  
0     A small fishing village must procure a local d...      Amazon Prime  
1     A Metro Family decides to fight a Cyber Crimin...      Amazon Prime  
2     After a man discovers his wife is cheating on ...      Amazon Prime  
3     Pink breaks the mold once again, bringing her ...      Amazon Prime  
4     Teenage Matt Banting wants to work with a famo...      Amazon Prime  
...                                                 ...               ...  
3068  The 23rd century adventures of Captain James T...              Hulu  
3069  Catapulted into the distant sector of the gala...              Hulu  
3070  Seventeen-year-old Paul is haunted by apocalyp...              Hulu  
3071  Rod Serling's seminal anthology series focused...              Hulu  
3072  The devastation is unleashed in the span of se...              Hulu  

[22998 rows x 11 columns]

Como visto acima, a removação foi bem sucedida.

### Eliminando valores nulos

Nosso objetivo agora é eliminar trechos do conjunto de dados com informações não preenchidas.

Não preencheremos os valores faltantes com "moda" ou "mediana" pelo fato de essa solução não se aplicar considerando o contexto de nosso DataFrame.

Preencher as linhas faltantes do atributo "director (diretor)" com a moda, por exemplo, não faz sentido considerando que temos inúmeros diretores diferentes que trabalharam em seus respectivos filmes e séries por diversos motivos diferentes.

Com isso em mente, iremos, agora, visualizar a quantidade de dados faltantes no DataFrame utilizando os métodos "isnull" e "sum".

In [ ]:
dados_mesclados.isnull().sum()

show_id                  0
type                     0
title                    0
country              11499
date_added            9554
release_year             0
rating                 864
duration               482
listed_in                0
description              4
dataset_de_origem        0
dtype: int64

Como mostrado acima, nosso conjunto de dados mesclado possui diversas linhas com valores nulos, principalmente a coluna "country".

Considerando que esse DataFrame contém uma quantidade de linhas considerável, se removermos todas as linhas com dados faltantes, ainda teremos muitas amostras restantes no conjunto.

Faremos isso através do método "dropna", da biblioteca "pandas".

In [ ]:
dados_mesclados = dados_mesclados.dropna()

Checando se a remoção foi bem sucedida:

In [ ]:
dados_mesclados.isnull().sum()

show_id              0
type                 0
title                0
country              0
date_added           0
release_year         0
rating               0
duration             0
listed_in            0
description          0
dataset_de_origem    0
dtype: int64

Verificando quantas linhas restaram:

In [ ]:
len(dados_mesclados)

10481

De 22998 linhas, restaram 10481.

Aproximadamente metade das linhas foram removidas.

### Checando por linhas duplicadas

Nesta breve etapa, iremos verificar se existem linhas duplicadas no nosso conjunto de dados mesclado.

Faremos isso utilizando os métodos "duplicated" e "sum" abaixo.

In [ ]:
dados_mesclados.duplicated().sum()

0

Como visto acima, não existem linhas duplicadas em nosso DataFrame.

### Removendo a coluna "show_id"

O atributo "show_id" é um identificador para cada uma das amostras de filmes e séries de tv, como observado abaixo.

Todavia, como o índice do dataset já faz o mesmo papel, iremos remover a coluna "show_id".

In [ ]:
dados_mesclados.head()

show_id     type                 title        country      date_added  \
1         s2    Movie  Take Care Good Night          India  March 30, 2021   
5197   s5198  TV Show        The Family Man          India    May 19, 2021   
5600   s5601  TV Show                 Bosch  United States    June 3, 2021   
5744   s5745  TV Show        Making the Cut  United States   June 11, 2021   
5920   s5921  TV Show          Sergio Ramos          Spain   June 18, 2021   

      release_year rating   duration                     listed_in  \
1             2018    13+    110 min          Drama, International   
5197          2021    13+  2 Seasons  Action, Drama, International   
5600          2021    16+  7 Seasons       Action, Drama, Suspense   
5744          2021    13+  2 Seasons  Special Interest, Unscripted   
5920          2021    16+  2 Seasons            Sports, Unscripted   

                                            description dataset_de_origem  
1     A Metro Family decides to fight a Cyber Crimin...      Amazon Prime  
5197  Srikant Tiwari has quit TASC and works in the ...      Amazon Prime  
5600  When a ten-year-old girl dies in an arson fire...      Amazon Prime  
5744  In the second season of Making the Cut, Heidi ...      Amazon Prime  
5920  Legends define themselves for who they have be...      Amazon Prime

Faremos isso através do método "drop".

In [ ]:
dados_mesclados = dados_mesclados.drop("show_id", axis=1)

In [ ]:
dados_mesclados.head()

type                 title        country      date_added  \
1       Movie  Take Care Good Night          India  March 30, 2021   
5197  TV Show        The Family Man          India    May 19, 2021   
5600  TV Show                 Bosch  United States    June 3, 2021   
5744  TV Show        Making the Cut  United States   June 11, 2021   
5920  TV Show          Sergio Ramos          Spain   June 18, 2021   

      release_year rating   duration                     listed_in  \
1             2018    13+    110 min          Drama, International   
5197          2021    13+  2 Seasons  Action, Drama, International   
5600          2021    16+  7 Seasons       Action, Drama, Suspense   
5744          2021    13+  2 Seasons  Special Interest, Unscripted   
5920          2021    16+  2 Seasons            Sports, Unscripted   

                                            description dataset_de_origem  
1     A Metro Family decides to fight a Cyber Crimin...      Amazon Prime  
5197  Srikant Tiwari has quit TASC and works in the ...      Amazon Prime  
5600  When a ten-year-old girl dies in an arson fire...      Amazon Prime  
5744  In the second season of Making the Cut, Heidi ...      Amazon Prime  
5920  Legends define themselves for who they have be...      Amazon Prime

Ao observar os resultados acima, a remoção foi bem sucedida.

### Consertando a coluna de índice.

Na etapa anterior, foi possível observar que a quantidade de linhas no DataFrame e seus índices não condizem.

Para consertarmos isso, iremos redefinir os valores de índice do conjunto de dados.

In [ ]:
dados_mesclados =  dados_mesclados.reset_index(drop=True)

Verificando o resultado:

In [ ]:
display(dados_mesclados)

type                 title        country         date_added  \
0        Movie  Take Care Good Night          India     March 30, 2021   
1      TV Show        The Family Man          India       May 19, 2021   
2      TV Show                 Bosch  United States       June 3, 2021   
3      TV Show        Making the Cut  United States      June 11, 2021   
4      TV Show          Sergio Ramos          Spain      June 18, 2021   
...        ...                   ...            ...                ...   
10476  TV Show            Heroic Age          Japan    August 14, 2009   
10477  TV Show  Black Blood Brothers          Japan  February 10, 2009   
10478  TV Show   Doogie Howser, M.D.  United States       July 1, 2008   
10479  TV Show         Lost in Space  United States       June 1, 2008   
10480  TV Show          Hikaru no Go          Japan      July 14, 2006   

       release_year rating   duration                              listed_in  \
0              2018    13+    110 min                   Drama, International   
1              2021    13+  2 Seasons           Action, Drama, International   
2              2021    16+  7 Seasons                Action, Drama, Suspense   
3              2021    13+  2 Seasons           Special Interest, Unscripted   
4              2021    16+  2 Seasons                     Sports, Unscripted   
...             ...    ...        ...                                    ...   
10476          2007  TV-PG   1 Season               Action, Adventure, Anime   
10477          2006  TV-MA   1 Season               Action, Adventure, Anime   
10478          1989  TV-PG  4 Seasons                  Comedy, Drama, Family   
10479          1965   TV-G  3 Seasons            Action, Adventure, Classics   
10480          2006  TV-14   1 Season  Anime, International, Science Fiction   

                                             description dataset_de_origem  
0      A Metro Family decides to fight a Cyber Crimin...      Amazon Prime  
1      Srikant Tiwari has quit TASC and works in the ...      Amazon Prime  
2      When a ten-year-old girl dies in an arson fire...      Amazon Prime  
3      In the second season of Making the Cut, Heidi ...      Amazon Prime  
4      Legends define themselves for who they have be...      Amazon Prime  
...                                                  ...               ...  
10476  Long ago, the advanced Tribe of Gold vanished ...              Hulu  
10477  It’s been a decade since the sacred war betwee...              Hulu  
10478  Douglas Howser is a doctor. He is also a 16-ye...              Hulu  
10479  Created by legendary producer Irwin Allen ("Th...              Hulu  
10480  Hikaru Shindo is like any sixth-grader in Japa...              Hulu  

[10481 rows x 10 columns]

O índice de nossas amostras agora está consertado.

### Transformando a coluna "duration" em duas colunas: "duration" e "seasons"

Em nosso conjunto de dados, existe uma categoria chamada "type (tipo)" a qual pode possuir um de dois valores: Movie (Filme) ou TV Show (Série de TV), como demonstrado abaixo.

In [ ]:
dados_mesclados["type"].value_counts()

Movie      6991
TV Show    3490
Name: type, dtype: int64

O problema disso é que TV Show e Movies não possuem seus valores na coluna "duration" medidos da mesma forma.

Movies tem duração apresentada em minutos.

TV Show tem sua duração apresentada em "Seasons (Temporadas)".

Diante desse cenário, iremos criar uma coluna apenas para duração em minutos e outra para quantidade de temporadas.

A coluna que guarda a quantidade de temporadas irá se chamar "seasons".

Para colocar nessa coluna apenas as temporadas, iremos aplicar uma expressão lambda na coluna "duration" para transformar em 0 todos os valores que não tiverem a substring "Season" neles. O resto dos valores, os quais são os valores que queremos, permanecerão os mesmos.

In [ ]:
dados_mesclados["seasons"] = dados_mesclados["duration"].apply(lambda x: 0 if not "Season" in x else x)

Verificando se a coluna foi adicionada:

In [ ]:
dados_mesclados.head()

type                 title        country      date_added  release_year  \
0    Movie  Take Care Good Night          India  March 30, 2021          2018   
1  TV Show        The Family Man          India    May 19, 2021          2021   
2  TV Show                 Bosch  United States    June 3, 2021          2021   
3  TV Show        Making the Cut  United States   June 11, 2021          2021   
4  TV Show          Sergio Ramos          Spain   June 18, 2021          2021   

  rating   duration                     listed_in  \
0    13+    110 min          Drama, International   
1    13+  2 Seasons  Action, Drama, International   
2    16+  7 Seasons       Action, Drama, Suspense   
3    13+  2 Seasons  Special Interest, Unscripted   
4    16+  2 Seasons            Sports, Unscripted   

                                         description dataset_de_origem  \
0  A Metro Family decides to fight a Cyber Crimin...      Amazon Prime   
1  Srikant Tiwari has quit TASC and works in the ...      Amazon Prime   
2  When a ten-year-old girl dies in an arson fire...      Amazon Prime   
3  In the second season of Making the Cut, Heidi ...      Amazon Prime   
4  Legends define themselves for who they have be...      Amazon Prime   

     seasons  
0          0  
1  2 Seasons  
2  7 Seasons  
3  2 Seasons  
4  2 Seasons

In [ ]:
dados_mesclados["seasons"].value_counts()

0             6991
1 Season      1965
2 Seasons      631
3 Seasons      319
4 Seasons      176
5 Seasons      136
6 Seasons       76
7 Seasons       59
8 Seasons       36
9 Seasons       21
10 Seasons      19
11 Seasons      12
12 Seasons      10
15 Seasons       4
13 Seasons       4
14 Seasons       3
17 Seasons       3
16 Seasons       3
19 Seasons       2
23 Seasons       2
22 Seasons       2
20 Seasons       1
32 Seasons       1
30 Seasons       1
25 Seasons       1
34 Seasons       1
29 Seasons       1
26 Seasons       1
Name: seasons, dtype: int64

A operação foi bem sucedida.

Agora, removeremos todos os valores que indicam quantidade de temporadas da coluna "duration" pois a mesma só irá registrar duração em minutos de filmes a partir de agora.

In [ ]:
dados_mesclados["duration"] = dados_mesclados["duration"].apply(lambda x: 0 if "Season" in x else x)

Verificando se a remoção foi bem sucedida:

In [ ]:
dados_mesclados["duration"].value_counts()

0          3490
90 min      184
93 min      182
94 min      182
97 min      168
           ... 
224 min       1
189 min       1
229 min       1
175 min       1
191 min       1
Name: duration, Length: 205, dtype: int64

Podemos ver acima que o valor 0 aparecee 3490 vezes.

3490 é exatamente o número de amostras de "TV Show" que nosso conjunto de dados contém.

Isso significa que 3490 valores da coluna "duration", os quais eram números referentes a quantidades de temporadas, foram removidos dessa coluna.

### Removendo a substring "min" dos valores da coluna "duration"

Prosseguindo, iremos remover a substring "min", presente nos valores da coluna "duration"

Faremos isso para facilitar futuras operações envolvendo esses valores.

Utilizando o método "apply", criaremos uma expressão lambda a qual usará o método "replace" para substituir a subtring "min", presente nos valores, por uma string vazia.

Caso a expressão lambda encontre um valor 0 (ou seja, um valor que não contém a substring "min") ela o deixará como está.

In [ ]:
dados_mesclados["duration"] = dados_mesclados["duration"].apply(lambda x: 0 if x == 0 else x.replace(" min", ''))

Verificando se tudo ocorreu como esperado:

In [ ]:
dados_mesclados["duration"].value_counts()

0      3490
90      184
93      182
94      182
97      168
       ... 
224       1
189       1
229       1
175       1
191       1
Name: duration, Length: 205, dtype: int64

Como é visível acima, a substring "min" sumiu dos valores.

Por fim, converteremos o tipo dessa coluna de para "inteiro".

In [ ]:
dados_mesclados["duration"] = dados_mesclados["duration"].astype("int")

### Removendo as substrings "Season" e "Seasons" dos valores da coluna "seasons"

Pelo mesmo motivo que removemos as substrings "min" na etapa anterior, iremos também remover "Season" e "Seasons", substrings presentes na coluna "seasons".

Utilizando o método "apply", criaremos uma expressão lambda a qual usará o método "replace" para substituir as subtrings "Season" e "Seasons", presente nos valores, por uma string vazia.

Caso a expressão lambda encontre um valor 0 (ou seja, um valor que não contém as substrings citadas) ela o deixará como está.

Primeiro, iremos remover a substring "Seasons":

In [ ]:
dados_mesclados["seasons"] = dados_mesclados["seasons"].apply(lambda x: 0 if x == 0 else x.replace("Seasons", ""))

Agora, removeremos a substring "Season":

In [ ]:
dados_mesclados["seasons"] = dados_mesclados["seasons"].apply(lambda x: 0 if x == 0 else x.replace("Season", ""))

Conferindo se foram removidas:

In [ ]:
dados_mesclados["seasons"].value_counts()

0      6991
1      1965
2       631
3       319
4       176
5       136
6        76
7        59
8        36
9        21
10       19
11       12
12       10
15        4
13        4
14        3
17        3
16        3
19        2
23        2
22        2
20        1
32        1
30        1
25        1
34        1
29        1
26        1
Name: seasons, dtype: int64

Também, converteremos o tipo dessa coluna para "inteiro".

In [ ]:
dados_mesclados["seasons"] = dados_mesclados["seasons"].astype("int")

### Transformando a coluna "type" em duas colunas: "movie" e "tv_series"

Ainda visando facilitar a visualização e operações futuras, iremos dividir o atributo "type" em duas colunas as quais guardarão apenas valores 0 ou 1.

A primeira coluna se chamará "movie" e guardará valores 1 caso a amostra seja um "Movie" e 0 caso seja uma "TV Show".

A segunda coluna se chamará "tv_series" e guardará valores 1 caso a amostra seja uma "TV Show" e 0 caso seja "Movie".

Criaremos elas através, também, de expressões lambda.

In [ ]:
dados_mesclados["movie"] = dados_mesclados["type"].apply(lambda x: 1 if "Movie" in x else 0)

In [ ]:
dados_mesclados["tv_series"] = dados_mesclados["type"].apply(lambda x: 1 if "TV" in x else 0)

Verificando se as operações funcionaram:

In [ ]:
dados_mesclados["movie"].value_counts()

1    6991
0    3490
Name: movie, dtype: int64

In [ ]:
dados_mesclados["tv_series"].value_counts()

0    6991
1    3490
Name: tv_series, dtype: int64

Considerando que tínhamos 3490 "TV Shows" no nosso conjunto de dados, percebemos acima que a coluna "tv_series" possui sua quantidade de valores iguais a 1 também em 3490.

Após isso, removeremos completamente a coluna "type" do DataFrame pois não a usaremos mais.

In [ ]:
dados_mesclados = dados_mesclados.drop("type", axis=1)

Conferindo os resultados:

In [ ]:
dados_mesclados.head()

title        country      date_added  release_year rating  \
0  Take Care Good Night          India  March 30, 2021          2018    13+   
1        The Family Man          India    May 19, 2021          2021    13+   
2                 Bosch  United States    June 3, 2021          2021    16+   
3        Making the Cut  United States   June 11, 2021          2021    13+   
4          Sergio Ramos          Spain   June 18, 2021          2021    16+   

   duration                     listed_in  \
0       110          Drama, International   
1         0  Action, Drama, International   
2         0       Action, Drama, Suspense   
3         0  Special Interest, Unscripted   
4         0            Sports, Unscripted   

                                         description dataset_de_origem  \
0  A Metro Family decides to fight a Cyber Crimin...      Amazon Prime   
1  Srikant Tiwari has quit TASC and works in the ...      Amazon Prime   
2  When a ten-year-old girl dies in an arson fire...      Amazon Prime   
3  In the second season of Making the Cut, Heidi ...      Amazon Prime   
4  Legends define themselves for who they have be...      Amazon Prime   

   seasons  movie  tv_series  
0        0      1          0  
1        2      0          1  
2        7      0          1  
3        2      0          1  
4        2      0          1

A coluna "type" foi removida com sucesso.

### Transformando a coluna "date added" para a mesma salvar apenas o ano

Nesta última seção de modificação, alteraremos o formato dos dados guardados pela coluna "date_added".

Essa coluna registra o dia, mês e ano em que um filme ou série foi adicionado à plataforma de streaming a qual ele pertence.

Contudo, faremos ela registrar apenas o ano por dois fatores:

1. Não iremos utilizar o dia e mês de lançamento de obras em nossas análises.
2. Fazer essa coluna guardar apenas o ano facilitará os cálculos que faremos em cima dela.

Começaremos criando uma lista para guardar todos os anos de todas as datas em que as obras foram adicionadas à plataformas de streaming.

In [ ]:
lista_de_anos = []

Após isso, iteraremos por todas as linhas da coluna "date_added", coletaremos apenas os anos através da função "split", salvaremos o valor na variável "ano" e adicionaremos esse valor na lista de anos.

In [ ]:
for linha in dados_mesclados["date_added"]:
  mes, ano = linha.split(", ")
  lista_de_anos.append(int(ano))

Para transformar a coluna, trocaremos todos os valores dela pelos valores salvos na lista de anos.

In [ ]:
dados_mesclados["date_added"] = lista_de_anos

No final, alteraremos o nome da coluna para "year_added" a fim de facilitar o entendimento.

In [ ]:
dados_mesclados.rename(columns={"date_added": "year_added"}, inplace=True)

## Salvando o conjunto de dados mesclado

Última visualização dos dados:

In [ ]:
dados_mesclados

title        country  year_added  release_year rating  \
0      Take Care Good Night          India        2021          2018    13+   
1            The Family Man          India        2021          2021    13+   
2                     Bosch  United States        2021          2021    16+   
3            Making the Cut  United States        2021          2021    13+   
4              Sergio Ramos          Spain        2021          2021    16+   
...                     ...            ...         ...           ...    ...   
10476            Heroic Age          Japan        2009          2007  TV-PG   
10477  Black Blood Brothers          Japan        2009          2006  TV-MA   
10478   Doogie Howser, M.D.  United States        2008          1989  TV-PG   
10479         Lost in Space  United States        2008          1965   TV-G   
10480          Hikaru no Go          Japan        2006          2006  TV-14   

       duration                              listed_in  \
0           110                   Drama, International   
1             0           Action, Drama, International   
2             0                Action, Drama, Suspense   
3             0           Special Interest, Unscripted   
4             0                     Sports, Unscripted   
...         ...                                    ...   
10476         0               Action, Adventure, Anime   
10477         0               Action, Adventure, Anime   
10478         0                  Comedy, Drama, Family   
10479         0            Action, Adventure, Classics   
10480         0  Anime, International, Science Fiction   

                                             description dataset_de_origem  \
0      A Metro Family decides to fight a Cyber Crimin...      Amazon Prime   
1      Srikant Tiwari has quit TASC and works in the ...      Amazon Prime   
2      When a ten-year-old girl dies in an arson fire...      Amazon Prime   
3      In the second season of Making the Cut, Heidi ...      Amazon Prime   
4      Legends define themselves for who they have be...      Amazon Prime   
...                                                  ...               ...   
10476  Long ago, the advanced Tribe of Gold vanished ...              Hulu   
10477  It’s been a decade since the sacred war betwee...              Hulu   
10478  Douglas Howser is a doctor. He is also a 16-ye...              Hulu   
10479  Created by legendary producer Irwin Allen ("Th...              Hulu   
10480  Hikaru Shindo is like any sixth-grader in Japa...              Hulu   

       seasons  movie  tv_series  
0            0      1          0  
1            2      0          1  
2            7      0          1  
3            2      0          1  
4            2      0          1  
...        ...    ...        ...  
10476        1      0          1  
10477        1      0          1  
10478        4      0          1  
10479        3      0          1  
10480        1      0          1  

[10481 rows x 12 columns]

Para futuras análises, as quais ocorrerão em outro notebook, iremos converter o dataset criado neste documento para um arquivo ".csv".

In [ ]:
dados_mesclados.to_csv("dados_mesclados_atualizado.csv")